src/Console/LintCommand.php

```diff
@@ -2,96 +2,100 @@
-            } else {
-                $finder = new Finder();
-                $found = iterator_to_array($finder->in($path)->exclude($exclude)->name('*.twig'));
-                if (!empty($found)) {
-                    $files = array_merge($files, $found);
-                }
-            }
-        }
+        $twig = $container['twig'];
+        $validator = $container['validator'];
+        $resolver = new ConfigurationResolver($container, getcwd(), [
+            'path' => $input->getArgument('path'),
+            'path-mode' => $input->getOption('path-mode'),
+            'reporter-service-name' => $input->getOption('reporter'),
+            'exclude' => $input->getOption('exclude'),
+            'severity' => $input->getOption('severity'),
+            'ruleset-class-name' => $input->getOption('ruleset'),
+            'config' => $input->getOption('config'),
+        ]);
+        $severityLimit = $resolver->getSeverityLimit();
+        $finder = $resolver->getFinder();
+        $reporter = $resolver->getReporter();
 
         $violations = [];
 
-        $ruleset = $input->getOption('ruleset');
+        $finderSanitizer = new FinderSanitizer();
+        $files = $finderSanitizer->sanitize($finder);
+//        $files = $finder;
 
-        if (!class_exists($ruleset)) {
-            throw new \InvalidArgumentException(sprintf('Ruleset class %s does not exist', $ruleset));
-        }
+        // TODO: should be refactored to a Runner class
+        foreach ($files as $file) {
+            /** @var SmartFileInfo $file */
+//            var_dump(
+//                'getRelativeFilePath',
+//                $file->getRelativeFilePath(),
+//                $file->getRelativePathname(),
+//                ''
+//                'getRelativeFilePathFromCwd',
+//                $file->getRelativeFilePathFromCwd(),
+//                'getRelativePathname',
+//                $file->getRelativePathname(),
+//                'getRelativeFilePathFromDirectory',
+//                $file->getRelativeFilePathFromDirectory(getcwd())
+//            );
 
-        if (!is_subclass_of($ruleset, RulesetInterface::class)) {
-            throw new \InvalidArgumentException('Ruleset class must implement '.RulesetInterface::class);
-        }
 
-        foreach ($files as $file) {
-            $violations = array_merge($violations, $container['validator']->validate(new $ruleset(), $container['twig']->tokenize(new \Twig\Source(
+            $ruleset = $resolver->getRuleset($file->getRelativePathname());
+            $currentViolations = $validator->validate($ruleset, $twig->tokenize(new \Twig\Source(
                 file_get_contents($file->getRealPath()),
                 $file->getRealPath(),
-                str_replace(realpath($path), rtrim($path, '/'), $file->getRealPath())
-            ))));
+                // TODO: relative path is without base dir if multiple in's are given.
+//                $file->getRelativePathname(),
```

Comment:
Commented out code again (see comment above). Also broken coding style in my opinion.